harness/determined/core/_context.py

```diff
@@ -242,9 +242,10 @@ def init(
 
     # We are on the cluster.
     cert = certs.default_load(info.master_url)
-    session = authentication.login_with_cache(info.master_url, cert=cert).with_retry(
-        util.get_max_retries_config()
-    )
+    session = authentication.login_from_task(
+        master_address=info.master_url,
+        cert=cert,
+    ).with_retry(util.get_max_retries_config())
```

Comment:
will this still work for init of core api in detached mode? think we have to fall back to user auth when not in a task

Reply:
i think so? detached mode checks if it's on cluster or not, and if it is, it uses this `login_from_task`. if not, it uses the session from the sdk client. in `_make_v2_context`

Reply:
ah yes, makes sense